### PR TITLE
Rename the parameter name to avoid mistaking it for the new operator, resolve #16416

### LINKED ISSUE #16416: [Bug] Compilation error when enabling both Bluetooth and pointer device
## Describe the Bug

In `rules.mk`, with `BLUETOOTH_ENABLE = yes` and `POINTING_DEVICE_ENABLE = yes` (i.e. with both Bluetooth and pointer device enabled) e.g.
```mk
BLUETOOTH_ENABLE = yes
BLUETOOTH_DRIVER = AdafruitBLE # or RN42

POINTING_DEVICE_ENABLE = yes
POINTING_DEVICE_DRIVER = analog_joystick
```

will cause a compilation error:

```
Compiling: drivers/bluetooth/adafruit_ble.cpp                                                      In file included from quantum/quantum.h:216,
                 from platforms/avr/drivers/analog.h:20,
                 from drivers/bluetooth/adafruit_ble.cpp:13:
quantum/pointing_device.h:80:56: error: expected ',' or '...' before 'new'
 bool           has_mouse_report_changed(report_mouse_t new, report_mouse_t old);
                                                        ^~~
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:456: .build/obj_kb_default/drivers/bluetooth/adafruit_ble.o] Error 1
Make finished with errors
make: *** [Makefile:478: kb:default] Error 1
```

## System Information

 - Keyboard:  Unreleased keyboard in testing (Using Pro Micro)
 - Operating system: Windows 10
 - AVR GCC version: 8.4.0

 - ARM GCC version: 10.1.0

 - QMK Firmware version: 0.15.16

 - Any keyboard related software installed? none
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

## Additional Context

The error refers to this file: [pointing_device.h](https://github.com/qmk/qmk_firmware/blob/ae5f43072f0de93b65854af569ba90564b8003de/quantum/pointing_device.h#L80)

I think the compiler is mistaking the parameter name "new" for "new operator", which causes this error.

I tried changing the name of the parameter from `new` to `newMouseReport`, and the above problem was solved. I think changing the parameter name to `newMouseReport` can also make it consistent with the other functions (`pointing_device_set_report(report_mouse_t newMouseReport)`).

### PR DESCRIPTION
**Pull request by ziteh**
## Description

The original parameter name `new` could be mistaken for new operator and cause a compilation error.

I changed this parameter name from `new` to `newMouseReport`,  I think changing the parameter name to newMouseReport can also make it consistent with the other functions (`pointing_device_set_report(report_mouse_t newMouseReport)`).

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fiexs #16416 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
There is no `new` operator in C.

**Comment by ziteh** [comment]
I'm also a little confused and unsure.

But "the compiler mistook it for a new operator" is the only reason I can think of that would cause an ` error: expected ',' or '...' before 'new'` in the compilation.

In addition, I renamed the parameters to `new_report` and `old_report`.

**Comment by fauxpark** [comment]
Sorry, I didn't see the linked issue.

It looks like `pointing_device.h` is being included in `adafruit_ble.cpp` (which, yes, is one of the few things written in C++ in QMK) through `analog.h` -> `quantum.h`. I doubt `analog.h` really needs most of the stuff in `quantum.h` - could you try changing it to `gpio.h` instead?

File: quantum/pointing_device.c [hunk 1]
```diff
@@ -77,8 +77,8 @@ extern const pointing_device_driver_t pointing_device_driver;
  * @param[in] old report_mouse_t
```
Comment (fauxpark) [inline quantum/pointing_device.c:77]: Doc comments should be changed to match.

File: quantum/pointing_device.h [hunk 2]
```diff
@@ -80,7 +80,7 @@ void           pointing_device_task(void);
 void           pointing_device_send(void);
 report_mouse_t pointing_device_get_report(void);
 void           pointing_device_set_report(report_mouse_t newMouseReport);
```
Comment (fauxpark) [inline quantum/pointing_device.h:82]:
```suggestion
void           pointing_device_set_report(report_mouse_t report);
```
\+ on line 42 of the .c file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by ziteh** [comment]
I tried to replace `quantum.h` in `analog.h` to `gpio.h`, but when using joystick as a pointer device (`POINTING_DEVICE_DRIVER = analog_joystick` in `rules.mk`), it causes a compilation error: 

```
Compiling: drivers/sensors/analog_joystick.c                                                       drivers/sensors/analog_joystick.c: In function 'axisToMouseComponent':
drivers/sensors/analog_joystick.c:67:44: error: implicit declaration of function 'abs' [-Werror=implicit-function-declaration]
         return percent * maxCursorSpeed * (abs(coordinate) / speedRegulator);
                                            ^~~
drivers/sensors/analog_joystick.c: In function 'analog_joystick_read':
drivers/sensors/analog_joystick.c:76:9: error: implicit declaration of function 'timer_elapsed' [-Werror=implicit-function-declaration]
     if (timer_elapsed(lastCursor) > ANALOG_JOYSTICK_READ_INTERVAL) {
         ^~~~~~~~~~~~~
drivers/sensors/analog_joystick.c:77:22: error: implicit declaration of function 'timer_read'; did you mean 'adc_read'? [-Werror=implicit-function-declaration]
         lastCursor = timer_read();
                      ^~~~~~~~~~
                      adc_read
cc1.exe: all warnings being treated as errors
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:457: .build/obj_kb_default/drivers/sensors/analog_joystick.o] Error 1
```

Although including `stdlib.h` and `timer.h` can solve the above problem, I'm not sure if this is a reasonable architecture.

And I found a lot of files that include `analog.h`. I'm not sure if there are other problems in other cases.

File: quantum/pointing_device.h [hunk 3]
```diff
@@ -79,8 +79,8 @@ void           pointing_device_init(void);
 void           pointing_device_task(void);
 void           pointing_device_send(void);
 report_mouse_t pointing_device_get_report(void);
-void           pointing_device_set_report(report_mouse_t newMouseReport);
-bool           has_mouse_report_changed(report_mouse_t new, report_mouse_t old);
+void           pointing_device_set_report(report_mouse_t report);
```
Comment (fauxpark) [inline quantum/pointing_device.h:82]: Needs to be changed in the .c file too.

**Comment by drashna** [comment]
Could the docs be updated? 

Also, linter wants changes.

**Comment by ziteh** [comment]
Sorry, I'm not sure what "Docs" actually means

And lint, which seems to be something related to CI ?

**Comment by fauxpark** [comment]
The lint error can be ignored. And docs refers to...the docs.

**Comment by ziteh** [comment]
Oh, I forgot that there is also a function list in the docs,
I think this is what you are talking about

File: docs/feature_pointing_device.md [hunk 4]
```diff
@@ -276,14 +276,14 @@ The report_mouse_t (here "mouseReport") has the following properties:
 To manually manipulate the mouse reports outside of the `pointing_device_task_*` functions, you can use:
 
 * `pointing_device_get_report()` - Returns the current report_mouse_t that represents the information sent to the host computer
-* `pointing_device_set_report(report_mouse_t newMouseReport)` - Overrides and saves the report_mouse_t to be sent to the host computer
+* `pointing_device_set_report(report_mouse_t mous_report)` - Overrides and saves the report_mouse_t to be sent to the host computer
```
Comment (drashna) [inline docs/feature_pointing_device.md:279]:
```suggestion
* `pointing_device_set_report(report_mouse_t mouse_report)` - Overrides and saves the report_mouse_t to be sent to the host computer
```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**